src/vmtest.rs

```diff
@@ -113,29 +109,18 @@ impl Vmtest {
 
     /// Setups up a `Qemu` instance for a run
     fn setup_qemu(&self, idx: usize, updates: Sender<Output>) -> Result<Qemu> {
-        let target = self
+        let mut target: Target = self
             .config
             .target
             .get(idx)
-            .ok_or_else(|| anyhow!("idx={} out of range", idx))?;
-        let image = self.resolve_path(target.image.as_deref());
-        let kernel = self.resolve_path(target.kernel.as_deref());
-        let rootfs = self.resolve_path(Some(target.rootfs.as_path())).unwrap();
-        let bios = self.resolve_path(target.vm.bios.as_deref());
-
-        Qemu::new(
-            updates,
-            image.as_deref(),
-            kernel.as_deref(),
-            target.kernel_args.as_ref(),
-            rootfs.as_path(),
-            bios.as_deref(),
-            &target.command,
-            &self.base,
-            target.uefi,
-            &target.vm,
-        )
-        .context("Failed to setup QEMU")
+            .ok_or_else(|| anyhow!("idx={} out of range", idx))?
+            .clone();
```

Comment:
Unnecessary clone?

Reply:
Without the clone we get a &Target.... which we later modify to adjust paths, and this is a can of worm to get right :)
It may work, but likely make the code more complicated than the cost of a clone of the config.

```
$ cargo build
   Compiling vmtest v0.9.0 (/home/chantra/devel/danobi-vmtest)
error[E0308]: `?` operator has incompatible types
   --> src/vmtest.rs:112:34
    |
112 |           let mut target: Target = self
    |  __________________________________^
113 | |             .config
114 | |             .target
115 | |             .get(idx)
116 | |             .ok_or_else(|| anyhow!("idx={} out of range", idx))?;
    | |________________________________________________________________^ expected `Target`, found `&Target`
    |
    = note: `?` operator cannot convert from `&config::Target` to `config::Target`
help: consider using clone here
    |
116 |             .ok_or_else(|| anyhow!("idx={} out of range", idx))?.clone();
    |                                                                 ++++++++

For more information about this error, try `rustc --explain E0308`.
```

Reply:
Ack, that makes sense to me